02_process/src/prepData_fun.R

```diff
@@ -0,0 +1,28 @@
+# Prepare the data for plotting
+
+
+prepData <- function(data_inPath, savePath) {
+  
+  require(dplyr)
+  require(readr)
+  require(stringr)
+  
+  eval_data <- readr::read_csv(data_inPath, col_types = 'iccd') %>%
+    filter(str_detect(exper_id, 'similar_[0-9]+')) %>%
+    mutate(col = case_when(
+      model_type == 'pb' ~ '#1b9e77',
+      model_type == 'dl' ~'#d95f02',
+      model_type == 'pgdl' ~ '#7570b3'
+    ), pch = case_when(
+      model_type == 'pb' ~ 21,
+      model_type == 'dl' ~ 22,
+      model_type == 'pgdl' ~ 23
+    ), n_prof = as.numeric(str_extract(exper_id, '[0-9]+')))
+  
+  saveRDS(eval_data, savePath)
```

Comment:
Here, you could just return the data rather than save as an RDS. You would just need to save the output from the function to pass into the next one, a la `eval_data <- prepData(data_inPath = "01_fetch/out/model_RMSEs.csv")`. The course will take you through edits that do that later, so just know that it is possible to return R objects from a function without writing them to files!

Reply:
Yep, that's usually what I do when I make a function, but I wasn't sure in this situation since the next function is in a different folder/phase...

Reply:
Yeah, good question. We can pass either files or R objects between the different phases. There will be examples of that as you continue!